### DISCUSSION TITLE
Errors installing 13.1 on Synology Container Manager

### DISCUSSION THREAD

**Discussion by sw90036** [alexjustesen/speedtest-tracker#870] [Q&A]
Hello -

This is an OUTSTANDING app, thank you for creating this!

I had speedtest-tracker running for ~6 month and recently, probably due to a recently Synology OS update, I was no longer able to login to the console, however the container was running and I was still getting email alerts.  

This evening I noticed there was a new version available, 13.1, so I deleted the old container and image and tried installing the new version, and it does appear to install however there now appears to be an issue with binding TCP ports or something else related to the web server and I am out of my depth here.

I have attached the log here, hoping this is something easy to fix.

Any advice here?

THANK YOU!

[speedtest-tracker.txt](https://github.com/alexjustesen/speedtest-tracker/files/13059744/speedtest-tracker.txt)

**Comment by sw90036** [comment]
Update:  I noted that the database.sqlite file was updated this evening when I did one of the first few rounds of installing but since I don't really need to keep history, I decided to delete the **.env** and **database.sqlite** files from the /config directory and tried the installation again.

This time I am getting a completey different set of error messages, log is attached.

I note here that one of the last messages before the container shuts down references **laravel-automations** -

     unable to start service laravel-automations: command exited 1
 
I googled this and see that it is a PHP framework but I'm not sure what I need to do in order to fix this. . 

ugh.


[speedtest-tracker_log2.txt](https://github.com/alexjustesen/speedtest-tracker/files/13059813/speedtest-tracker_log2.txt)

**Reply by alexjustesen** [reply]
So new `.env` assumes you're using `mysql` or `mariadb` as the database. If you want to use SQLite use **ONLY** the following `DB_` environment variables.

```
DB_CONNECTION=sqlite
DB_HOST=/config/database.sqlite
```

**Reply by sw90036** [reply]
Hi Alex -

Thank you for the reply!  I added those two variables into the startup configuration 

![speedtest-tracker_config](https://github.com/alexjustesen/speedtest-tracker/assets/44692292/379e850e-ff92-4243-ab96-d8facb68bf2f)

I am not getting new set of messages during startup

[sppedtest-tracker_log3.txt](https://github.com/alexjustesen/speedtest-tracker/files/13075498/sppedtest-tracker_log3.txt)

From the terminal, here are the active listening ports

![speedtest-tracker_listeners](https://github.com/alexjustesen/speedtest-tracker/assets/44692292/6cf7a1c0-1f47-4b5e-9a68-be36af888c61)

And active processes

![speedtest-tracker_processes](https://github.com/alexjustesen/speedtest-tracker/assets/44692292/ac5f29cb-5910-4d0f-b3f0-0cb08d670723)


Any other suggestions for me to try?

**Reply by alexjustesen** [reply]
If you're referring to the fpm error try rebooting the container, sometimes that can occur randomly and I haven't been able to reproduce it on my DS920+.

Oh and also make sure you have an `APP_KEY` set.

**Comment by SzymonRivera** [comment]
It also stopped working for me after updating from version 0.11.17 but i got it working again (Synology DS 918+).

I cleared the config folder, added the 2 DB_ variables and changed the BUILD_PHP_VERSION variable to 8.1.

**Comment by sw90036** [comment]
I've been fiddling around with this and with every effort, it's one step forward and one step back.

Based on suggestions here, I added the two DB_ variables, set PHP_BUILD_VERSION to 8.1 and generated/added the APP_KEY variable.  I re-run setup and the container comes up this time without any error messages in the log but the web server still does not respond on all three of the configured ports.

One thing I notice now is that the .ENV and DATABASE.SQLITE files are being created in the top level folder, not the CONFIG folder, even though I specify the config folder during setup, and I even tried configuring the container folder for \docker\speedtest2\ but setup will not continue unless I specify CONFIG as the subfolder.

![Untitled](https://github.com/alexjustesen/speedtest-tracker/assets/44692292/143613d2-b882-4c03-af2b-1a4d7390273b)

I see in the documentation that SQLite is supported by Laravel as a database driver but it is no longer the recommended driver for this project, v0.11.22 dropped SQLite support.

SQLITE has worked great for me in the past but I'm certainly happy to switch to MySQL or PostgreSQL I just don't how to do that unless it's a part of the standard setup & configure process.

I've spent too much time on this so I think I'm going to put this on hold until there are updated instructions on how to setup this up on Synology.

**Comment by sw90036** [comment]
I just tried the installation again with the current build, which I believe is v0.19.0, and I still have all of the same issues which I am unable to resolve.  I decided to try installing an earlier version and was able to successfully install v0.11.10 and it came up right away so I will stick with that for now.

Is there any possibility of updating the documentation for Synology Container Manager for the current version?  If not, I am ok to stick with the old version.

Again, this is an OUTSTANDING app, thank you!

**Reply by svenvg93** [reply]
The latest version is 0.20.7 can you give it a try with the needed environment variables listed here in the docker compose example: https://docs.speedtest-tracker.dev/getting-started/installation/installation

**Reply by SzymonRivera** [reply]
I'm also on version 0.19.0 at the moment and for me it still works. I don't remember why, but I had PHP set to 8.1 but I see that only PHP 7.4 and 8.0 are installed. Here are my settings, hope it helps you.

![php](https://github.com/user-attachments/assets/5d15efa3-8d0a-4979-a96b-2c89c7899e5e)
![settings 1](https://github.com/user-attachments/assets/66356192-62d4-4202-a4a1-8924e74b9828)
![settings 2](https://github.com/user-attachments/assets/569d87e7-0aa6-41ef-9fc7-f4619478663a)

**Reply by svenvg93** [reply]
Im not known with Synology Container Manager, this qeustion might be stupid 😅. 
Are all those Env needed for Synology?  As mine compose only needs the env for the application, not for PHP etc

```
    speedtest-tracker:
        image: lscr.io/linuxserver/speedtest-tracker:0.20.8
        container_name: speedtest-tracker
        restart: unless-stopped
        environment:
          TZ: Europe/Amsterdam 
          APP_TIMEZONE: Europe/Amsterdam 
          DISPLAY_TIMEZONE: Europe/Amsterdam
          DB_CONNECTION: sqlite
          APP_KEY: ${APP_KEY}
          APP_URL: stt.${DOMAIN}
          SPEEDTEST_PING_URL: 1.1.1.1 # Optional
          SPEEDTEST_SCHEDULE: 6 */2 * * *
          SPEEDTEST_SERVERS: 52365
          PRUNE_RESULTS_OLDER_THAN: 0
          DATETIME_FORMAT: "j M Y, G:i:s"
          CHART_DATETIME_FORMAT: "j/m G:i"
```

**Reply by alexjustesen** [reply]
@SzymonRivera keep in mind the PHP version installed into Synology OS doesn't affect the container's version. That being said with the LSIO images it's running PHP 8.3: https://github.com/linuxserver/docker-speedtest-tracker/blob/main/Dockerfile.

@svenvg93 Synology adds a lot of default values, I think the newer versions of container manager clean this up a bit.
